### ISSUE TITLE
Include or don't include an http request implementation

### ISSUE THREAD

**Issue by a-s-o** [motorcode/relm#8]
Alternatives:
- superagent (currently used in tutorials)
- axios (similar to angular $http)
- request
- fetch API polyfill

Why:
- if an implementation is included, then we can create helpers/standardized api for async actions

**Comment by SeeThruHead** [comment]
fetch with the option to swap out for other promise based http libs?

**Comment by a-s-o** [comment]
Sure, but then there is a need to also include a promise polyfill at least for a little while. I don't mind that but in that case, I would like to see if that is not a turn off for people.

I will start looking at some alternatives and see how much they change the build size.

**Comment by SeeThruHead** [comment]
I think any http lib that people might want to use in a modern stack is promise based.

**Comment by a-s-o** [comment]
Agreed; I hardly work without them now. So, I take it you are in the **include** an http lib camp instead of leaving it as a user decision.

BTW I am leaning towards your direction but just a bit indecisive. Thanks for your input.

**Comment by SeeThruHead** [comment]
Yeah I would opt for making this as user friendly as possible. That means setting sane defaults for most (maybe all?) basic features like http requests. I do however think that you should be able to supply your own http lib / promise polyfill as an override. Just my 2cents
